src/services/import-service.js

```diff
@@ -139,6 +141,63 @@ class ImportService {
 
     return batch;
   }
+
+  async getProgressForBatch(batch) {
+    if (batch.completedOn) {
+      return 1;
+    }
+
+    const countGroups = await this.taskStore.toAggregateArray([
+      {
+        $match: {
+          batchId: batch._id
+        }
+      }, {
+        $group: {
+          _id: '$processed',
+          count: {
+            $sum: 1
+          }
+        }
+      }
+    ]);
+
+    const stats = countGroups.reduce((accumulator, current) => {
+      accumulator.totalCount += current.count;
+      if (current._id === true) {
```

Comment:
I'm not sure I get this condition, is `current._id` not the task id?

Reply:
Fixed, it is indeed a bit confusing, I also needed a few seconds to remember what that is so :)).